mempool/v0/clist_mempool.go

```diff
@@ -183,13 +183,18 @@ func (mem *CListMempool) TxsFront() *clist.CElement {
 	return mem.txs.Front()
 }
 
-// GetTxByKey is not supported by the v0 mempool but it is required to satisfy the mempool interface.
+// GetTxByKey retrieves a transaction from the mempool using its key.
 func (mem *CListMempool) GetTxByKey(key types.TxKey) (types.Tx, bool) {
-	return nil, false
+	e, ok := mem.txsMap.Load(key)
+	if !ok {
+		return nil, false
+	}
+	memTx, ok := e.(*clist.CElement).Value.(*mempoolTx)
+	return memTx.tx, ok
 }
 
-// IsTxEvicted is not supported by the v0 mempool but it is required to satisfy the mempool interface.
-func (mem *CListMempool) IsTxEvicted(key types.TxKey) bool {
+// WasRecentlyEvicted returns false consistently as this implementation does not support transaction eviction.
```

Comment:
[question] why does this implementation not support transaction eviction?

Reply:
evicted txs are valid transactions that were evicted because of size/ttl reasons for example when mempool is full. afaik Clist mempool on the other hand directly rejects transactions if the cache is full rather than evicting them.

Reply:
There is also no TTL on the v0 mempool